dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -162,93 +171,100 @@ instance Arbitrary Directory where
-                , ( 7, lift1 BoolLit)
-                , ( 1, lift2 BoolAnd)
-                , ( 1, lift2 BoolOr)
-                , ( 1, lift2 BoolEQ)
-                , ( 1, lift2 BoolNE)
-                , ( 1, lift3 BoolIf)
-                , ( 1, lift0 Natural)
-                , ( 7, fmap NaturalLit arbitrary)
-                , ( 1, lift0 NaturalFold)
-                , ( 1, lift0 NaturalBuild)
-                , ( 1, lift0 NaturalIsZero)
-                , ( 1, lift0 NaturalEven)
-                , ( 1, lift0 NaturalOdd)
-                , ( 1, lift0 NaturalToInteger)
-                , ( 1, lift0 NaturalShow)
-                , ( 1, lift2 NaturalPlus)
-                , ( 1, lift2 NaturalTimes)
-                , ( 1, lift0 NaturalSubtract)
-                , ( 1, lift0 Integer)
-                , ( 7, fmap IntegerLit integer)
-                , ( 1, lift0 IntegerShow)
-                , ( 1, lift0 Double)
-                , ( 7, lift1 DoubleLit)
-                , ( 1, lift0 DoubleShow)
-                , ( 1, lift0 Text)
-                , ( 1, lift1 TextLit)
-                , ( 1, lift2 TextAppend)
-                , ( 1, lift0 List)
-                , let listLit = do
-                          n  <- Test.QuickCheck.choose (0, 3)
-                          xs <- Test.QuickCheck.vectorOf n arbitrary
-                          let ys = Data.Sequence.fromList xs
-                          ListLit <$> arbitrary <*> pure ys
-
-                  in  ( 1, listLit)
-                , ( 1, lift2 ListAppend)
-                , ( 1, lift0 ListBuild)
-                , ( 1, lift0 ListFold)
-                , ( 1, lift0 ListLength)
-                , ( 1, lift0 ListHead)
-                , ( 1, lift0 ListLast)
-                , ( 1, lift0 ListIndexed)
-                , ( 1, lift0 ListReverse)
-                , ( 1, lift0 Optional)
-                , ( 1, lift0 OptionalFold)
-                , ( 1, lift0 OptionalBuild)
-                , ( 1, lift1 Record)
-                , ( 7, lift1 RecordLit)
-                , ( 1, lift1 Union)
-                , ( 7, lift2 Combine)
-                , ( 1, lift2 CombineTypes)
-                , ( 7, lift2 Prefer)
-                , ( 1, lift3 Merge)
-                , ( 1, lift2 ToMap)
-                , ( 7, lift2 Field)
-                , ( 7, lift2 Project)
-                , ( 1, lift1 Assert)
-                , ( 1, lift2 Equivalent)
-                , ( 7, lift1 Embed)
-                , ( 7, lift2 ImportAlt)
-                ]
-            )
+            (Generic.Random.genericArbitraryRecG customGens weights)
             standardizedExpression
+      where
+        customGens
+            :: Gen Integer    -- Generates all Integer fields in Expr
+            :+ Gen Text.Text  -- Generates all Text fields in Expr
+            :+ ()
+        customGens =
+               integer
+               -- We generate "_"s mostly to check code operating on 'Lam's and 'Pi's
```

Comment:
Perhaps expand upon this explanation to mention that variables named `_` are encoded differently

Reply:
I expanded the comment a bit.